lib/livebook_web/live/session_live/k8s_runtime_component.ex

```diff
@@ -0,0 +1,952 @@
+          socket
+          |> assign(home_pvc: nil, pvc_action: nil)
+          |> pvc_options()
+
+        {:ok, %{body: %{"message" => message}}} ->
+          assign_nested(socket, :pvc_action, error: message)
+      end
+
+    {:noreply, socket}
+  end
+
+  def handle_event("cancel_delete_pvc", %{}, socket) do
+    {:noreply, assign(socket, pvc_action: nil)}
+  end
+
+  def handle_event("init", %{}, socket) do
+    config = build_config(socket)
+    runtime = Runtime.K8s.new(config, socket.assigns.reqs.pod)
+    Session.set_runtime(socket.assigns.session.pid, runtime)
+    Session.connect_runtime(socket.assigns.session.pid)
+    {:noreply, socket}
+  end
+
+  def handle_event("open_save_config", %{}, socket) do
+    changeset = config_secret_changeset(socket, %{name: @config_secret_prefix})
+    save_config = %{changeset: changeset, inflight: false, error: false}
+    {:noreply, assign(socket, save_config: save_config)}
+  end
+
+  def handle_event("cancel_save_config", %{}, socket) do
+    {:noreply, assign(socket, save_config: nil)}
+  end
+
+  def handle_event("validate_save_config", %{"secret" => secret}, socket) do
+    changeset =
+      socket
+      |> config_secret_changeset(secret)
+      |> Map.replace!(:action, :validate)
+
+    {:noreply, assign_nested(socket, :save_config, changeset: changeset)}
+  end
+
+  def handle_event("save_config", %{"secret" => secret}, socket) do
+    changeset = config_secret_changeset(socket, secret)
+
+    case Ecto.Changeset.apply_action(changeset, :insert) do
+      {:ok, secret} ->
+        {:noreply, save_config_secret(socket, secret, changeset)}
+
+      {:error, changeset} ->
+        {:noreply, assign_nested(socket, :save_config, changeset: changeset)}
+    end
+  end
+
+  def handle_event("load_config", %{"name" => name}, socket) do
+    secret = Enum.find(socket.assigns.hub_secrets, &(&1.name == name))
+
+    case Jason.decode(secret.value) do
+      {:ok, config_defaults} ->
+        {:noreply,
+         socket
+         |> assign(config_defaults: config_defaults)
+         |> load_config_defaults()}
+
+      {:error, _} ->
+        {:noreply, socket}
+    end
+  end
+
+  @impl true
+  def handle_async(:rbac_check, {:ok, %{errors: errors, permissions: permissions}}, socket) do
+    status = if errors === [], do: :ok, else: :errors
+    {:noreply, assign(socket, :rbac, %{status: status, errors: errors, permissions: permissions})}
+  end
+
+  def handle_async(:load_namespace_options, {:ok, [:ok, {:ok, resp}]}, socket) do
+    socket =
+      case resp do
+        %Req.Response{status: 200, body: %{"items" => resources}} ->
+          namespace_options = Enum.map(resources, & &1["metadata"]["name"])
+
+          socket
+          |> assign(:namespace_options, namespace_options)
+          |> set_namespace(List.first(namespace_options))
+          |> assign(:cluster_check, %{status: :ok, error: nil})
+
+        %Req.Response{status: _other} ->
+          # cannot list namespaces
```

Comment:
Should we show an error?

Reply:
It is technically not necessary to list namespaces as long as you have permission to create a pod in your target namespace. So what I do in the UI in this case: Show a text input field instead of a select for the target namespace.

Reply:
Sounds good!